driver/src/test/java/org/neo4j/driver/internal/net/SocketClientTest.java

```diff
@@ -22,32 +22,69 @@
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 
 import java.io.IOException;
 import java.net.ServerSocket;
 import java.nio.ByteBuffer;
 import java.nio.channels.ByteChannel;
+import java.security.GeneralSecurityException;
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
 import java.util.List;
 
 import org.neo4j.driver.internal.logging.DevNullLogger;
 import org.neo4j.driver.internal.security.SecurityPlan;
 import org.neo4j.driver.v1.exceptions.ClientException;
 import org.neo4j.driver.v1.exceptions.ServiceUnavailableException;
 
+import static junit.framework.TestCase.fail;
 import static org.hamcrest.CoreMatchers.equalTo;
 import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.runners.Parameterized.Parameters;
+import static org.mockito.Matchers.any;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 import static org.neo4j.driver.internal.net.BoltServerAddress.LOCAL_DEFAULT;
 
+@RunWith(Parameterized.class)
 public class SocketClientTest
 {
     private static final int CONNECTION_TIMEOUT = 42;
 
     @Rule
     public ExpectedException exception = ExpectedException.none();
 
+    @Parameters(name = "{0} connections")
+    public static Collection<Object[]> data()
+    {
+        return Arrays.asList(new Object[][] {
+                { "insecure", SecurityPlan.insecure() },
+                { "encrypted", createEncryptedSecurityPlan() }
```

Comment:
Can we just use `SecurityPlan.forAllCertificates()` here and declare `#data()` to throw `Exception`?